### PR TITLE
docs: Update Admin Partitions with more explicit commands by using shell variables

### PR DESCRIPTION
**Pull request by david-yu**
### Description

 Adding some additional polish to Admin Partitions docs by updating the versions and making use of environment variables. Also use namespaces when running `kubectl` commands as well. 

### Testing & Reproduction steps
* In the case of bugs, describe how to replicate
* If any manual tests were done, document the steps and the conditions to replicate
* Call out any important/ relevant unit tests, e2e tests or integration tests you have added or are adding

### Links
Include any links here that might be helpful for people reviewing your PR (Tickets, GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.

Please be mindful not to leak any customer or confidential information. HashiCorp employees may want to use our internal URL shortener to obfuscate links.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern
* [ ] checklist [folder](./../docs/config) consulted

### CODE REVIEW

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (curtbushko) [inline website/content/docs/enterprise/admin-partitions.mdx]: Is that "-n"supposed to be there? Or is it supposed to have a namespace value passed to it?

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (curtbushko) [inline website/content/docs/enterprise/admin-partitions.mdx]: Same with "-n" from here

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (david-yu) [reply website/content/docs/enterprise/admin-partitions.mdx]: Thanks I caught that as well!

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (david-yu) [reply website/content/docs/enterprise/admin-partitions.mdx]: Thanks!

**Review by curtbushko — APPROVED** [review]
Looks good!

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:100]:
```suggestion
- A designated Kubernetes `LoadBalancer` service must be exposed on the Consul server cluster. This enable the following communication channels to the Consul servers and the `default` partition:
  - RPC on port 8300 
  - Gossip on port 8301
  - HTTPS API requests on port 443 API requests 
- Mesh gateways must be deployed as a Kubernetes `LoadBalancer` service on port 443 across all Kubernetes clusters.
- Cross-partition networking must be implemented as described in [Cross-Partition Networking](#cross-partition-networking). 
```
Trying to make this more bullet-y list-y so it's easier to consume while still providing all of the information.  
  
This seems like a general requirement: 

- All Consul clients must be able to communicate with the Consul servers in the `default` partition, and all servers must be able to communicate with the clients. 

Maybe move it up the top? Or create a Networking sub requirement and put it there?

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:112]:
```suggestion
#### Prepare to install Consul across multiple Kubernetes clusters
```

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:134]:
```suggestion
  $ kubectl create secret --namespace consul generic license --from-file=key=./path/to/license.hclic
```
I think we want to always use the expanded form of the flags. Also, is `consul generic license` an actual value or is it placeholder text? We've been trying to use brackets (square or angle) for placeholder text

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:147]:
```suggestion
  $ kubectl create secret --namespace consul generic license --from-file=key=./path/to/license.hclic
```

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:201]:
```suggestion
  $ helm install ${HELM_RELEASE_SERVER} hashicorp/consul --version "0.43.0" --create-namespace --namespace consul --values server.yaml
```

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:207]:
```suggestion
  $ kubectl get services --selector="app=consul,component=server" --namespace consul --output jsonpath="{range .items[*]}{@.status.loadBalancer.ingress[*].ip}{end}"
```

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:222]:
```suggestion
  $ kubectl get secret ${HELM_RELEASE_SERVER}-consul-ca-cert --context ${SERVER_CONTEXT} -n consul --output yaml | kubectl apply --namespace consul --context ${CLIENT_CONTEXT} --filename -
```

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:228]:
```suggestion
  $ kubectl get secret ${HELM_RELEASE_SERVER}-consul-ca-key --context ${SERVER_CONTEXT} --namespace consul --output yaml | kubectl apply --namespace consul --context ${CLIENT_CONTEXT} --filename -
```

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:234]:
```suggestion
  $ kubectl get secret ${HELM_RELEASE_SERVER}-consul-partitions-acl-token --context ${SERVER_CONTEXT} --namespace consul --output yaml | kubectl apply --namespace consul --context ${CLIENT_CONTEXT} --filename -
```

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:239]:
```suggestion
1. Switch to the workload client clusters:
```

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:306]:
```suggestion
  $ helm install ${HELM_RELEASE_CLIENT} hashicorp/consul --version "0.43.0" --create-namespace --namespace consul --values client.yaml 
```

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:322]:
```suggestion
  $ kubectl get secret --namespace consul ${HELM_RELEASE_SERVER}-consul-bootstrap-acl-token --template "{{ .data.token | base64decode }}"
```

**Review by trujillo-adam — COMMENTED** [review]
Added some comments, questions, and suggestions

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (david-yu) [reply website/content/docs/enterprise/admin-partitions.mdx:100]: We can move up higher as a general networking requirement under its own section.

**Comment by david-yu** [comment]
@trujillo-adam This PR is ready for re-review.

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:71]: I image that the next question from users is going to be 'well, how do I do that?' This requirement is fulfilled by configuring firewalls in whatever infrastructure environment, correct? Not Consul-specific settings? Should we add something like "Refer to the documentation for your infrastructure environment for instructions on how to configure routes between nodes." Or am I misunderstanding?

**Review by trujillo-adam — COMMENTED** [review]
There are still a few instances where we're using `-n` instead of `--namespace`, which may not be that important. The bigger question is if we're leaving users hanging with the General Network Requirements. IDK

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (david-yu) [reply website/content/docs/enterprise/admin-partitions.mdx:71]: I expanded, we actually do want to make sure clients also can communicate to servers via RPC, Gossip and HTPS as well for general networking requirements.

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (trujillo-adam) [inline website/content/docs/enterprise/admin-partitions.mdx:76]:
```suggestion
All Consul clients must be able to communicate with the Consul servers in the `default` partition. All servers must also be able to communicate with the clients.

For Consul on Kubernetes, deploy a dedicated `partition` Kubernetes `LoadBalancer` service (refer to [Kubernetes Requirements](#kubernetes-requirements) for additional information). 

For other runtimes, refer to the documentation for your infrastructure environment for instructions on how to allow communication on the following ports:
- 8300 (RPC)  
- 8301 (gossip)
- 443 (HTTPS API requests)  
```
I think we can point readers to the dedicated section.

**Review by trujillo-adam — COMMENTED** [review]
Sorry for all the back and forth. Almost there :)

**Review by trujillo-adam — APPROVED** [review]
👍

**Comment by david-yu** [comment]
Still waiting on the Core team for their feedback around general requirements. Will merge after ensuring that the section looks accurate.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/671663.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 4a2b61a1f4708eefd2cc97fa6786e93a4b763dc6 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 4a2b61a1f4708eefd2cc97fa6786e93a4b763dc6 onto `release/1.12.x` succeeded!

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (blake) [inline website/content/docs/enterprise/admin-partitions.mdx:73]: It might be the phrasing that is throwing me off, but my initial thought from reading this sentence is that it is not technically correct. It is true that client and server agents need to be able to communicate with each other, but the clients will communicate with the servers over the gossip pool for its assigned partition, not the `default` partition.

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (blake) [inline website/content/docs/enterprise/admin-partitions.mdx:109]: I think "…and the `default` partition" should be removed since the client's config determines the partition it is trying to join.

File: website/content/docs/enterprise/admin-partitions.mdx
Comment (blake) [inline website/content/docs/enterprise/admin-partitions.mdx:143]: You might be able to simplify this a bit by removing the `use-context` commands and instead passing the `--context` flag to each of the `kubectl create` commands. E.g., `kubectl create secret --context $CLIENT_CONTEXT …`

**Pull request merged**
